### PR TITLE
Switch to DLL builds on windows

### PR DESCRIPTION
**Pull request by greglandrum**
We've had a long-open issue on the rdkit side that log messages (used for error reporting) do not show up properly on Windows: https://github.com/rdkit/rdkit/issues/2760
After some investigation it looks like this is only solvable by switching the windows builds of the python wrappers to use DLL builds of the RDKit core instead of statically linking it. The Mac and Linux builds have been dynamic linkage from the beginning, so they don't have this problem.

This PR is one-liner which switches on DLL builds for windows.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by greglandrum** [comment]
@conda-forge-admin, please rerender

**Comment by greglandrum** [comment]
@jaimergp, @mcs07: do either of you know how to get the recipe to copy the DLLs from the build/lib directory to wherever they should end up?
This could be an RDKit misconfiguration (in which case I will fix it upstream), but it looks like the cmake config for the DLLs  on windows should be the same as the shared libs on linux/mac, and those work fine with the conda recipe.

**Comment by jaimergp** [comment]
I think DLLs belong to LIBRARY_BIN or SCRIPTS (they need to be placed along the executables, basically, but it depends on how the package is configured). Try both and let's see which one works!

File: recipe/install_rdkit_dev.bat [hunk 1]
```diff
@@ -6,6 +6,9 @@ if not exist "%LIBRARY_INC%" mkdir %LIBRARY_INC%
 REM copy .lib files to LIBRARY_LIB
 copy lib\*.lib %LIBRARY_LIB%
 
+REM copy .dll files to LIBRARY_LIB
+copy lib\*.dll %LIBRARY_LIB%
```
Comment (jaimergp) [inline recipe/install_rdkit_dev.bat:10]: Let's try this one:

```suggestion
if not exist "%LIBRARY_BIN%" mkdir %LIBRARY_BIN%
copy lib\*.dll %LIBRARY_BIN%
```

**Comment by jaimergp** [comment]
Ah, there's another issue:

```
(base) D:\bld\rdkit_1635854592478\work>REM copy .dll files to LIBRARY_LIB 

(base) D:\bld\rdkit_1635854592478\work>copy lib\*.dll D:\bld\rdkit_1635854592478\_h_env\Library\lib 
lib\*.dll 
The system cannot find the file specified.
        0 file(s) copied.
```

It doesn't look like the DLLs are there :/

**Comment by greglandrum** [comment]
> It doesn't look like the DLLs are there :/

hmmm, let me take a look at that. It may need to be ../lib

**Comment by greglandrum** [comment]
from looking at the output of the build on windows in CI, it looks like there is stuff being installed into:
```
[ 37%] Linking CXX shared library ..\..\bin\RDKitGraphMol.dll
```
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=401362&view=logs&j=171a126d-c574-5c8c-1269-ff3b989e923d&t=1183ba29-a0b5-5324-8463-2a49ace9e213&l=5939

and:
```
-- Installing: D:/bld/rdkit_1635854592478/_h_env/Library/lib/RDKitGraphMol.dll
```
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=401362&view=logs&j=171a126d-c574-5c8c-1269-ff3b989e923d&t=1183ba29-a0b5-5324-8463-2a49ace9e213&l=8806

That second one looks like the conda install is putting the DLL in Library/lib, but I think the issue is that these dll files need to be in Library/bin (I guess that's $LIBRARY_BIN). Looking at a local environment on windows, the boost DLLs are in Library/bin and the corresponding .lib files are in Library/lib. The RDKit should be analogous.

**Pull request merged**